### PR TITLE
CSHARP-5343: Encryption-related tests failing locally

### PR DESCRIPTION
**Pull request by sanych-sun**

### CODE REVIEW

File: src/MongoDB.Driver.Encryption/MongoDB.Driver.Encryption.csproj [hunk 1]
```diff
@@ -13,100 +13,102 @@
...
-  </PropertyGroup>
+  <Target Name="DownloadNativeBinaries_MacOS" BeforeTargets="BeforeBuild" Condition="!Exists('$(MSBuildProjectDirectory)/runtimes/osx/native/libmongocrypt.dylib')">
+    <PropertyGroup>
+      <LibMongoCryptSourceUrl>https://mciuploads.s3.amazonaws.com/libmongocrypt-release/macos/r1.11/9a88ac5698e8e3ffcd6580b98c247f0126f26c40/libmongocrypt.tar.gz</LibMongoCryptSourceUrl>
+      <LibMongoCryptSourcePath>lib/libmongocrypt.dylib</LibMongoCryptSourcePath>
+      <LibMongoCryptPackagePath>runtimes/osx/native</LibMongoCryptPackagePath>
+    </PropertyGroup>
 
-  <Target Name="DownloadNativeBinaries_linux_x64" BeforeTargets="BeforeBuild" Condition="!Exists('$(MSBuildProjectDirectory)/linux/x64/libmongocrypt.so')">
+    <MSBuild Projects ="$(MSBuildProjectFullPath)"
```
Comment (sanych-sun) [inline src/MongoDB.Driver.Encryption/MongoDB.Driver.Encryption.csproj:23]: So this is where the main magic is situated: we call MSBuild to start "another" project, which is actually the same project (`$(MSBuildProjectFullPath)`) by explicitly invoking Target named `DownloadNativeBinary` and pass some variables that define where the file should be pulled from. Also important stuff: we set `TargetFramework=once`, this is to override default Targets we have. If we do not override that - the target will be started 3 times (once per existing target framework).

Comment (sanych-sun) [reply src/MongoDB.Driver.Encryption/MongoDB.Driver.Encryption.csproj:23]: And yep - repeat for each file to download.

File: src/MongoDB.Driver.Encryption/MongoDB.Driver.Encryption.csproj [hunk 2]
```diff
@@ -13,100 +13,102 @@
...
-    <Exec Command="tar -zxvf $(IntermediateOutputPathLSF)windows/libmongocrypt.tar.gz --strip-components=1 -C $(IntermediateOutputPathLSF)windows bin/mongocrypt.dll" />
-    <Copy SourceFiles="$(IntermediateOutputPathLSF)windows/mongocrypt.dll" DestinationFolder="$(MSBuildProjectDirectory)/windows" Retries="5" RetryDelayMilliseconds="100"/>
+
+    <MSBuild Projects ="$(MSBuildProjectFullPath)"
+             Properties="TargetFramework=once;LibMongoCryptSourceUrl=$(LibMongoCryptSourceUrl);LibMongoCryptSourcePath=$(LibMongoCryptSourcePath);LibMongoCryptPackagePath=$(LibMongoCryptPackagePath)"
+             Targets="DownloadNativeBinary" />
   </Target>
 
-  <Target Name="DownloadNativeBinaries_Alpine" BeforeTargets="BeforeBuild" Condition="!Exists('$(MSBuildProjectDirectory)/linux/alpine/libmongocrypt.so')">
+  <Target Name="DownloadNativeBinary">
```
Comment (sanych-sun) [inline src/MongoDB.Driver.Encryption/MongoDB.Driver.Encryption.csproj:76]: This Target is the one that actually doing the work.

File: src/MongoDB.Driver.Encryption/MongoDB.Driver.Encryption.csproj [hunk 3]
```diff
@@ -13,100 +13,102 @@
...
+    <DownloadFile SourceUrl="$(LibMongoCryptSourceUrl)" DestinationFolder="$(LibMongoCryptTmpPath)"/>
+    <Exec Command="tar -zxvf $(LibMongoCryptTmpPath)/libmongocrypt.tar.gz -C $(LibMongoCryptTmpPath) $(LibMongoCryptSourcePath)" />
+    <Copy SourceFiles="$(LibMongoCryptTmpPath)/$(LibMongoCryptSourcePath)" DestinationFolder="$(MSBuildProjectDirectory)/$(LibMongoCryptPackagePath)" />
+    <RemoveDir Directories="$(LibMongoCryptTmpPath)" />
   </Target>
 
   <ItemGroup>
-    <Content Include="$(MSBuildProjectDirectory)/linux/x64/libmongocrypt.so">
-      <Link>libmongocrypt\linux-x64\libmongocrypt.so</Link>
+    <Content Include="$(MSBuildProjectDirectory)/runtimes/osx/native/libmongocrypt.dylib">
```
Comment (sanych-sun) [inline src/MongoDB.Driver.Encryption/MongoDB.Driver.Encryption.csproj:87]: I've changed where we store the file in project folder to match the library search path, so we do not need to set the env variable anymore. Path to the file in the package was not changed.

**Comment by adelinowona** [comment]
LGTM

**Pull request merged**